transformers/events/log_value/converter.go

```diff
@@ -0,0 +1,89 @@
+// VulcanizeDB
+// Copyright © 2018 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package log_value
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/factories/event"
+	"github.com/vulcanize/vulcanizedb/pkg/core"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/eth"
+	"math/big"
+)
+
+type Converter struct {
+	db *postgres.DB
+}
+
+const Val event.ColumnName = "val"
+
+func (Converter) toEntities(contractAbi string, logs []core.HeaderSyncLog) ([]LogValueEntity, error) {
+	var entities []LogValueEntity
+	for _, log := range logs {
+		var entity LogValueEntity
+		address := log.Log.Address
+		abi, parseErr := eth.ParseAbi(contractAbi)
+		if parseErr != nil {
+			return nil, parseErr
+		}
+
+		contract := bind.NewBoundContract(address, abi, nil, nil, nil)
+		unpackErr := contract.UnpackLog(&entity, "LogValue", log.Log)
+		if unpackErr != nil {
+			return nil, unpackErr
+		}
+
+		entity.HeaderID = log.HeaderID
+		entity.LogID = log.ID
+		entities = append(entities, entity)
+	}
+
+	return entities, nil
+}
+
+func (c Converter) ToModels(abi string, logs []core.HeaderSyncLog) ([]event.InsertionModel, error) {
+	entities, entityErr := c.toEntities(abi, logs)
+	if entityErr != nil {
+		return nil, fmt.Errorf("converter couldn't convert logs to entities: %v", entityErr)
+	}
+
+	var models []event.InsertionModel
+	for _, logValueEntity := range entities {
+		bigIntVal := new(big.Int).SetBytes(logValueEntity.Val[:]) // TODO: verify this is a good type
```

Comment:
I wasn't quite sure what type to make the `val` value. It's not obvious from the contract what it's being used for, but since the OSM contracts are oracles (presumably about collateral evaluations), I figured an int would be a good choice, and the resulting `bigint` values don't seem too objectionable.

Reply:
Contract says `val uint128` so `big.Int` is fine :)

Reply:
This is true but in general I'm a fan of using https://github.com/makerdao/vdb-mcd-transformers/blob/staging/transformers/shared/utilities.go#L67 or https://github.com/makerdao/vdb-mcd-transformers/blob/staging/transformers/shared/utilities.go#L62 depending on whether it's an `int` or `uint`, since that distinction can cause sneaky 🐛

Reply:
Thanks 🤦‍♂ I always forget we have these nice conversion utils!